L1Trigger/DemonstratorTools/plugins/GTTFileReader.cc

```diff
@@ -48,61 +50,144 @@ class GTTFileReader : public edm::stream::EDProducer<> {
+                                   iConfig.getParameter<std::vector<std::string>>("filesOutputToCorrelator"),
+                                   l1t::demo::kFramesPerTMUXPeriod,
+                                   l1t::demo::kGTTBoardTMUX,
+                                   kEmptyFramesOutputToCorrelator_,
+                                   l1t::demo::kChannelMapOutputToCorrelator);
+    produces<l1t::VertexWordCollection>(l1VertexCollectionName_);
+  }
+  if (processInputTracks_) {
+    fileReaderInputTracks_ =
+        l1t::demo::BoardDataReader(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
+                                   iConfig.getParameter<std::vector<std::string>>("filesInputTracks"),
+                                   l1t::demo::kFramesPerTMUXPeriod,
+                                   l1t::demo::kGTTBoardTMUX,
+                                   kEmptyFramesInputTracks_,
+                                   l1t::demo::kChannelMapInput);
+    produces<TTTrackCollection>(l1TrackCollectionName_);
+  }
+  if (processOutputToGlobalTrigger_) {
+    // fileReaderOutputToGlobalTrigger_ =
+    //   l1t::demo::BoardDataReader(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
+    // 				 iConfig.getParameter<std::vector<std::string>>("filesOutputToGlobalTrigger"),
+    // 				 l1t::demo::kFramesPerTMUXPeriod,
+    // 				 l1t::demo::kGTTBoardTMUX,
+    // 				 kEmptyFramesOutputToGlobalTrigger_,
+    // 				 l1t::demo::kChannelMapInput);
+    throw std::invalid_argument("Processing OutputToGlobalTrigger files has not been fully implemented and validated.");
+    // need to produce output collections for Prompt and Displaced Jets, HTMiss, ETMiss, Taus, Mesons, Vertices, and Isolated Tracks
+  }
 }
 
 // ------------ method called to produce the data  ------------
 void GTTFileReader::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
   using namespace edm;
   using namespace l1t::demo::codecs;
-
-  l1t::demo::EventData eventData(fileReader_.getNextEvent());
-
-  l1t::VertexWordCollection vertices(decodeVertices(eventData.at({"vertices", 0})));
-
-  edm::LogInfo("GTTFileReader") << vertices.size() << " vertices found";
-
-  iEvent.put(std::make_unique<l1t::VertexWordCollection>(vertices));
+  if (processOutputToCorrelator_) {
+    l1t::demo::EventData correlatorEventData(fileReaderOutputToCorrelator_.getNextEvent());
+    l1t::VertexWordCollection vertices(decodeVertices(correlatorEventData.at({"vertices", 0})));
+    edm::LogInfo("GTTFileReader") << vertices.size() << " vertices found";
+
+    iEvent.put(std::make_unique<l1t::VertexWordCollection>(vertices), l1VertexCollectionName_);
+  }  // end if ( processOutputToCorrelator_ )
+
+  if (processInputTracks_) {
+    l1t::demo::EventData inputEventData(fileReaderInputTracks_.getNextEvent());
+    auto inputTracks = std::make_unique<TTTrackCollection>();
+    for (size_t i = 0; i < 18; i++) {
```

Comment:
Is `18` a defined/definable constant somewhere?

Reply:
It should be defined/definable indeed, I'll update that